dsp/transform/hilbert_test.go

```diff
@@ -0,0 +1,41 @@
+// Copyright ©2024 The Gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package transform
+
+import (
+	"fmt"
+	"math"
+	"slices"
+	"testing"
+)
+
+func TestHilbertAnalytic(t *testing.T) {
+	testCases := []struct {
+		input []float64
+	}{
+		{[]float64{0, 0, 0, 0}},
+		{[]float64{1, 2, 3, 4}},
+		{[]float64{1, 2, 3, 4, 5}},
+	}
```

Comment:
Please hoist this out of the function to be a `var hilbertAnalyticSignalTests []struct{ name string; input ...`.

Reply:
This feedback is confusing to me, as the FFT test cases are written inside the function. With the update I made for readability, is this still needed?

Reply:
Yes, I'd like the var outside the function.